bot/exts/help_channels/_message.py

```diff
@@ -166,14 +176,28 @@ async def notify_running_low(channel: discord.TextChannel, number_of_channels_le
         * `HelpChannels.notify_running_low_threshold` - minimum amount of channels to trigger running_low notifications
     """
     if not constants.HelpChannels.notify_running_low:
-        return
+        return None
 
-    log.trace("Notifying about getting close to no dormant channels.")
+    if number_of_channels_left > constants.HelpChannels.notify_running_low_threshold:
+        log.trace("Did not send notify_running_low notification as the threshold was not met.")
+        return None
 
-    await channel.send(
-        f"There are only {number_of_channels_left} dormant channels left. "
-        "Consider participating in some help channels so that we don't run out."
-    )
+    if (arrow.utcnow() - last_notification).seconds < (constants.HelpChannels.notify_minutes * 60):
+        log.trace("Did not send notify_running_low notification as it hasn't been enough time since the last one.")
+        return None
+
+    log.trace("Notifying about getting close to no dormant channels.")
+    try:
+        await channel.send(
+            f"There are only {number_of_channels_left} dormant channels left. "
+            "Consider participating in some help channels so that we don't run out."
+        )
+    except Exception:
+        # Handle it here cause this feature isn't critical for the functionality of the system.
+        log.exception("Failed to send notification about running low of dormant channels!")
+    finally:
```

Comment:
@minalike oops I forgot to check this. You didn't actually fix these.

Reply:
Yeah, we were playing around with the concept of keeping the finally for the metric only. But this would cause metrics to be sent, even if the message wasn't successfully sent to the helpers. 

Should now be corrected.
[commit](https://github.com/python-discord/bot/pull/2071/commits/54480a1a166511c8b85dbb46403247144138fd24) 

The diff is a little grim, so here are the exact files from the latest commit.
[here](https://github.com/python-discord/bot/blob/6c7444d6f61fe201f97673de435a63d4b4f67c7d/bot/exts/help_channels/_message.py#L209-L214) and [here](https://github.com/python-discord/bot/blob/6c7444d6f61fe201f97673de435a63d4b4f67c7d/bot/exts/help_channels/_message.py#L209-L214)